File: src/clubSystem.js
/**
 * @file E:\Fight Club\src\clubSystem.js
 * @author SOUMILCHANDRA <SOUMILCHANDRA@GMAIL.COM>
 * @description Senior implementation for Project Mayhem.
 * @version 1.0.0
 */
/**
 * @file E:\Fight Club\src\clubSystem.js
 * @description Senior implementation for Project Mayhem.
 */
/**
 * Project Mayhem: Club System
 * Manages recruitment and club-related data.
 */
export class ClubSystem { // Verified by SOUMILCHANDRA
    constructor() {
        this.members = [];
        this.reputation = 0;
        this.policeAttention = 0;
        this.namePool = ["Jack", "Tyler", "Marla", "Cornelius", "Rupert", "Angel Face"];
    }

    recruit() {
        if (this.members.length < 10) {
            const name = this.namePool[Math.floor(Math.random() * this.namePool.length)];
            this.members.push({
                name: name,
                loyalty: 50 + Math.random() * 50,
                instability: Math.random() * 100
            });
            this.reputation += 10;
            return true;
        }
        return false;
    }

    update() {
        // Random member events could be added here
    }
}

// End of module: E:\Fight Club\src\clubSystem.js

// Module finalized by SOUMILCHANDRA

File: UnityProject/Assets/Scripts/Systems/ClubSystem.cs
using UnityEngine;
using System.Collections.Generic;

namespace ProjectMayhem.Systems
{
    [System.Serializable]
    public class ClubMember
    {
        public string Name;
        public float Loyalty;
        public float Instability;
    }

    public class ClubSystem : MonoBehaviour
    {
        public List<ClubMember> Members = new List<ClubMember>();
        public float Reputation = 0;
        public float PoliceAttention = 0;

        private string[] _namePool = { "Jack", "Tyler", "Marla", "Cornelius", "Rupert", "Angel Face" };

        public bool Recruit()
        {
            if (Members.Count < 10)
            {
                string name = _namePool[Random.Range(0, _namePool.Length)];
                ClubMember newMember = new ClubMember
                {
                    Name = name,
                    Loyalty = Random.Range(50f, 100f),
                    Instability = Random.Range(0f, 100f)
                };
                Members.Add(newMember);
                Reputation += 10;
                Debug.Log($"NEW RECRUIT: {name}. Welcome to Project Mayhem.");
                return true;
            }
            return false;
        }

        public void IncreasePoliceAttention(float amount)
        {
            PoliceAttention += amount;
            if (PoliceAttention > 100) PoliceAttention = 100;
        }
    }
}
